packages/cosmos/src/client.rs

```diff
@@ -147,6 +148,74 @@ impl<Res> PerformQueryWrapper<Res> {
 }
 
 impl Cosmos {
+    async fn get_and_update_simulation_sequence(
+        &self,
+        address: Address,
+    ) -> Result<BaseAccount, Error> {
+        let mut guard = self.pool.get().await?;
+        let cosmos = guard.get_inner_mut();
+        let mut sequences = cosmos.simulate_sequences.write().await;
+        let sequence_info = sequences
+            .entry(address)
+            .or_insert_with(|| SequenceInformation {
+                sequence: 0,
+                timestamp: Instant::now(),
+            });
+        let mut base_account = self.get_base_account(address).await?;
+        let diff = Instant::now().duration_since(sequence_info.timestamp);
+        if diff.as_secs() <= 30 {
+            let sequence = std::cmp::max(sequence_info.sequence, base_account.sequence);
+            if sequence != sequence_info.sequence {
+                *sequence_info = SequenceInformation {
+                    sequence,
+                    timestamp: Instant::now(),
+                };
+            }
+            base_account.sequence = sequence;
+            return Ok(base_account);
+        }
+        Ok(base_account)
+    }
+
+    async fn update_broadcast_sequence(&self, address: Address) -> Result<(), Error> {
+        let mut guard = self.pool.get().await?;
+        let cosmos = guard.get_inner_mut();
+        let mut sequences = cosmos.broadcast_sequences.write().await;
+        sequences
+            .entry(address)
+            .and_modify(|item| item.sequence += 1);
```

Comment:
This looks a bit dangerous to me. We just broadcast a transaction, so we know the sequence number we used. Instead of incrementing the cache, which may be out of sync with the transaction we just made, I'd recommend taking it from the transaction itself.